compiler-core/src/javascript/typescript.rs

```diff
@@ -0,0 +1,530 @@
+//! This module is responsible for generating Typescript type declaration files.
+//! This code is run during the code generation phase along side the normal
+//! Javascript code emission. Here we walk through the typed AST and translate
+//! the Gleam statements into their Typescript equivalent. Unlike the Javascript
+//! code generation, the Typescript generation only needs to look at the module
+//! statements and not the expressions that may be _inside_ those statements.
+//! This is due to the Typescript declarations only caring about inputs and outputs
+//! rather than _how_ those outputs are generated.
+//!
+//! ## Links
+//! <https://www.typescriptlang.org/>
+//! <https://www.typescriptlang.org/docs/handbook/declaration-files/introduction.html>
+
+use std::{cell::RefCell, rc::Rc, sync::Arc};
+
+use heck::CamelCase;
+use itertools::Itertools;
+
+use crate::{
+    ast::{
+        Statement, TypedArg, TypedConstant, TypedExternalFnArg, TypedModule,
+        TypedRecordConstructor, TypedStatement,
+    },
+    docvec,
+    pretty::{break_, Document, Documentable},
+    type_::{Type, TypeVar},
+};
+
+use super::{concat, import::Imports, line, lines, wrap_args, Output};
+
+const INDENT: isize = 4;
```

Comment:
```suggestion
use super::{INDENT, concat, import::Imports, line, lines, wrap_args, Output};
```

Let's use the same as in the JS code